Tests/RxSwiftTests/PrimitiveSequenceTest.swift

```diff
@@ -792,17 +792,53 @@ extension PrimitiveSequenceTest {
             ])
     }
 
+    func testAsSingle_subscribeOnSuccess() {
+        var events: [SingleEvent<Int>] = []
+
+        _ = Single.just(1).subscribe(onSuccess: { element in
+            events.append(.success(element))
+        }, onError: { error in
+            events.append(.error(error))
+        })
+        
+        XCTAssertEqual(events, [.success(1)])
+    }
+
+    func testAsSingle_subscribeOnError() {
+        var events: [SingleEvent<Int>] = []
+
+        _ = Single.error(testError).subscribe(onSuccess: { element in
+                events.append(.success(element))
+            }, onError: { error in
+                events.append(.error(error))
+            })
+
+        XCTAssertEqual(events, [.error(testError)])
+    }
+
     #if TRACE_RESOURCES
-        func testAsSingleReleasesResourcesOnComplete() {
+        func testAsSingleReleasesResourcesOnComplete1() {
             _ = Observable<Int>.just(1).asSingle().subscribe({ _ in })
         }
 
+        func testAsSingleReleasesResourcesOnComplete2() {
+            _ = Observable<Int>.just(1).asSingle().subscribe(onSuccess: { _ in })
+        }
+
         func testAsSingleReleasesResourcesOnError1() {
-        _ = Observable<Int>.error(testError).asSingle().subscribe({ _ in })
+            _ = Observable<Int>.error(testError).asSingle().subscribe({ _ in })
         }
 
         func testAsSingleReleasesResourcesOnError2() {
-        _ = Observable<Int>.of(1, 2).asSingle().subscribe({ _ in })
+            _ = Observable<Int>.of(1, 2).asSingle().subscribe({ _ in })
+        }
+
+        func testAsSingleReleasesResourcesOnError3() {
```

Comment:
Yeah @kzaher , sure thing. 
I thought we would just want to test resources get released even when using the full subscription block at that case. No problem to get rid of those.